lib-py/casmutils/xtal/_xtal/structure-py.cxx

```diff
@@ -32,31 +32,40 @@ void to_poscar(const xtal::Structure& writeable, const std::string& filename)
     return;
 }
 
-void set_lattice(xtal::Structure* self, const xtal::Lattice& new_lattice, std::string mode)
+xtal::COORD_TYPE coord_modes_switch(std::string mode)
```

Comment:
I like this routine, but I think it needs a better name. Something as mundane as `string_to_coord_mode` would be more clear, but maybe you can come up with something better. The "switch" is just a way you chose to implement it, the caller doesn't know or care about that.